script.js

```diff
@@ -0,0 +1,184 @@
+            if (btn.id === 'cancel') {
+                display.textContent = '';
+                userEntry = [];
+            } else if (btn.id === 'backspace') {
+                display.textContent = display.textContent.slice(0, -1);
+                userEntry.pop();
+            } else {
+                const result = convertExpression(userEntry);
+                userEntry = result;
+                display.textContent = result.join('');
+            };
+        } else {
+            display.textContent += btn.textContent;
+            userEntry.push(btn.value);
+        };
+    });
+});
+
+function buttonColor(button) {
+    operBtns.forEach(btn => btn.style.color = 'white');
+
+    if (button.classList.contains('oper')) {
+        button.style.color = 'orange'
+    };
+};
+
+
+function add(numOne, numTwo) {
+    //prevents numbers being concatenated 
+    const a = Number(numOne);
+    const b = Number(numTwo);
+    const result = (a + b).toString()
+    return result;
+};
+
+function subtract(numOne, numTwo) {
+    return numOne - numTwo;
+};
+
+function multiply(numOne, numTwo) {
+    return numOne * numTwo;
+};
+
+function divide(numOne, numTwo) {
+    return numOne / numTwo;
+};
+
+function percentOf(numOne, numTwo) {
+    //divide by 100 to get 1% of numTwo
+    return (numTwo / 100) * numOne;
+}
+
+
+const oper = {
+    '+': {
+        fn: add,
+        precedence: 1,
+    },
+    '-': {
+        fn: subtract,
+        precedence: 1,
+    },
+    '*': {
+        fn: multiply,
+        precedence: 2,
+    },
+    '/': {
+        fn: divide,
+        precedence: 2,
+    },
+};
+
+//converts to postfix expression
+function convertExpression(expr) {
+    const output = [];
+    const stack = [];
+    let accum = '';
+
+    for (let i = 0; i < expr.length; i++) {
+        let top = stack.length - 1;
+        const char = expr[i];
+        const next = expr[i + 1];
+
+        if (isNaN(char) && char !== '.') {
+
+            if (!stack.length || char === '(' || stack[top] === '(') {
+                stack.push(char);
+            } else if (oper.hasOwnProperty(char)) {
+                while (stack.length && stack[top] !== '(' && oper[char].precedence <= oper[stack[top]].precedence) {
+                    output.push(stack.pop());
+                    top = stack.length - 1;
+                };
+                stack.push(char);
+            } else {
+                while (stack[top] !== '(') {
+                    output.push(stack.pop());
+                    top = stack.length - 1;
+                };
+                stack.pop(); //pops remaining '('
+            };
+        } else {
+
+            if (!isNaN(next) || next === '.') {
+                accum += char;
+            } else if (accum) {
+                accum += char;
+                output.push(accum);
+                accum = '';
+            } else {
+                output.push(char); //pushes single digit numbers
+            };
```

Comment:
here is a good example of my earlier mention of returning early, the 'else' is the only thing that takes place after the last condition, so if you do the equivalent of an early return in the last block, you don't need it. 

As you're in a loop, the equivalent is either break, or continue

break, breaks out the whole loop, you dont want this to happen here 

continue, continues to the next iteration of the loop, skipping all the code that follows it for this current iteration- this is the one you want